From 486d34b567bbf6ac488a0a8b43cbd3a00ea62237 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Tue, 14 Oct 2025 17:48:05 +0530
Subject: [PATCH 63/63] Fix for clearing outstanding requests

Recreated the below patch:

From e107bdc375aa23a4c0a3d23978cfe8b378bce09a Mon Sep 17 00:00:00 2001
From: Jayaprakash Mutyala <mutyalax.jayaprakash@intel.com>
Date: Mon, 20 Sep 2021 14:06:05 +0000
Subject: [PATCH] Fix for clearing outstanding requests

IPMB requests were made valid even if there was an error in opening the
mux bus. This fix makes the IPMB requests valid only we were able to
successfully open the Mux fd.

Tested:
Verified using busctl command.
Command: busctl call xyz.openbmc_project.Ipmi.Channel.Ipmb
         /xyz/openbmc_project/Ipmi/Channel/Ipmb org.openbmc.Ipmb
         sendRequest yyyyay 0 0x30 0 0x70 2 0 32
Response: (iyyyyay) 0 49 0 112 193 0

Signed-off-by: pandiarajt <pandiarajt@ami.com>
---
 ipmbbridged.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ipmbbridged.cpp b/ipmbbridged.cpp
index b9a60a3..d35778b 100644
--- a/ipmbbridged.cpp
+++ b/ipmbbridged.cpp
@@ -834,7 +834,6 @@ std::tuple<int, uint8_t, uint8_t, uint8_t, uint8_t, std::vector<uint8_t>>
                                 std::shared_ptr<IpmbRequest> request,
                                 const uint8_t pcieSlot)
 {
-    makeRequestValid(request);
     std::filesystem::path p = "/dev/i2c-mux/PCIE_Mux/Pcie_Slot_" +
                               std::to_string(pcieSlot);
 
@@ -962,7 +961,8 @@ std::tuple<int, uint8_t, uint8_t, uint8_t, uint8_t, std::vector<uint8_t>>
                                " To avoid log storm, not all the logs for the "
                                "issue will be shown: ";
                 }
-                if (!(requestAddLogCount % ipmbRequestAddLogInterval) && (requestAddLogCount != 0))
+                if (!(requestAddLogCount % ipmbRequestAddLogInterval) &&
+                    (requestAddLogCount != 0))
                 {
                     msgToLog = "requestAdd: There are " +
                                std::to_string(requestAddLogCount) +
-- 
2.34.1

